libbpf-tools/offcputime.bpf.c

```diff
@@ -107,4 +106,15 @@ int BPF_PROG(sched_switch, bool preempt, struct task_struct *prev, struct task_s
 	return 0;
 }
 
+SEC("tp_btf/sched_switch")
+int BPF_PROG(sched_switch_btf, bool preempt, struct task_struct *prev, struct task_struct *next) {
+	return handle_switch(ctx, prev);
+}
+
+SEC("kprobe/finish_task_switch")
+int BPF_PROG(sched_switch_tp) {
```

Comment:
Let's fallback to raw tracepoint instead. See https://github.com/iovisor/bcc/pull/4217.

Reply:
Thanks for replay, but on my test env, the `"raw_tp/sched_switch"` won't work:
```
libbpf: sched_switch is not found in vmlinux BTF                                                                                                               
    entries_lock                                                                                                                                               
failed to get syms                                                                                                                                             
    -                 (0)                                                                                                                                      
        51                                                                                                                                                     
                                                                                                                                                               
    entries_lock                                                                                                                                               
failed to get syms
    -                 (0)
        4811

    entries_lock        
```
The comm is empty and PID is always 0, I think the `prev` is not correctly taken. Do you have any suggest?

Reply:
> Let's fallback to raw tracepoint instead. See #4217.

I tried to make it work and succeed, but the call stacks still seem always belong to the tracepoint `raw_tp/sched_switch`.

Changed code:
```c
SEC("raw_tp/sched_switch")
int BPF_PROG(sched_switch_tp) {
	struct task_struct *prev = (struct task_struct *)((struct bpf_raw_tracepoint_args *)ctx)->args[1];
	return handle_switch(ctx, prev);
}
```
Output example:
```
    entries_lock                                                                                                                                    [250/15106]
    -                kworker/2:2 (2704182)                                                                                                                     
        9                                                                                                                                                      
                                                                                                                                                                       

    entries_lock                       
    wait4                              
    -                bash (2708107)
        13                             

    entries_lock                       
    -                ksoftirqd/0 (10)
        1                              
```

Reply:
Does #4217 work for you ?

Reply:
> Does #4217 work for you ?

Everything works fine except the callstacks, so close this pr until I find the reason and solutions。